src/plugin/nodeSpec.ts

```diff
@@ -299,11 +297,12 @@ const getDefaultParseDOMForLeafNode = (nodeName: string) => [
         return false;
       }
 
-      const attrs = {
-        fields: JSON.parse(dom.getAttribute("fields") ?? "{}") as unknown,
-      };
+      const maybeFieldAttrs = dom.getAttribute("fields");
+      const fields = maybeFieldAttrs
+        ? (JSON.parse(maybeFieldAttrs) as unknown)
+        : undefined;
```

Comment:
```suggestion
      const fields: unknown | undefined = maybeFieldAttrs && JSON.parse(maybeFieldAttrs)
```

Reply:
feel free to drop the last two commits if this suggestion is breaking stuff (tests look to be failing)

Reply:
Ah yep, this changes `undefined` to `null`, null values are serialised to html as the string `null` 🤔